### PR TITLE
external/avs-device-sdk: Integration improvement

### PR DESCRIPTION
**Pull request by zhouxinhe**
1. Use arastorage instead of SQLite3 in avs.
2. Implement MediaPlayer and VoiceRecorder.
3. Update avs_test example app.
@Taejun-Kwon @jsdosa @xixidodo

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [710a0237946d8af76c0784ebd1c4341efc81bba3] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [710a0237946d8af76c0784ebd1c4341efc81bba3] - Code Rule Check (C++) OK.

File: external/avs-device-sdk/AVSCommon/Utils/src/LibcurlUtils/LibcurlUtils.cpp [hunk 1]
```diff
@@ -104,15 +104,15 @@ bool prepareForTLS(CURL* handle) {
     }
 
 // Only allow disabling the verification of hosts and peers in debug configurations.
-#ifdef DEBUG
+//#ifdef DEBUG
```
Comment (jiawei-xiang) [inline external/avs-device-sdk/AVSCommon/Utils/src/LibcurlUtils/LibcurlUtils.cpp]: need to clean it?

File: apps/examples/avs_test/avs_test_main.cpp [hunk 2]
```diff
@@ -74,67 +76,208 @@
...
+	pthread_attr_t attr;
+	struct sched_param sparam;
+	sparam.sched_priority = 100;
+
+	pthread_attr_init(&attr);
+	pthread_attr_setstacksize(&attr, CONFIG_AVS_TEST_THREAD_STACKSIZE);
+	pthread_attr_setschedparam(&attr, &sparam);
+
+	int status;
+    pthread_t thread;
```
Comment (JeongJunSik) [inline apps/examples/avs_test/avs_test_main.cpp]: please check tab size

File: apps/examples/avs_test/avs_test_main.cpp [hunk 3]
```diff
@@ -74,67 +76,208 @@
...
+	status = pthread_create(&thread, &attr, avs_test_entry, NULL);
+	if (status != 0) {
+		printf("[%s] pthread_create failed, status=%d\n", __FUNCTION__, status);
+		return 0;
+	}
+
+	pthread_setname_np(thread, "avs_test_main");
+	pthread_join(thread, NULL);
+    return 0;
```
Comment (JeongJunSik) [inline apps/examples/avs_test/avs_test_main.cpp]: please check tab size

File: external/avs-device-sdk/AVSCommon/Utils/src/Logger/ConsoleLogger.cpp [hunk 4]
```diff
@@ -42,7 +45,11 @@ void ConsoleLogger::emit(
     const char* threadMoniker,
     const char* text) {
     std::lock_guard<std::mutex> lock(m_coutMutex);
+#ifdef NCOUT
+	syslog(LOG_ERR, "%s\n", m_logFormatter.format(level, time, threadMoniker, text).c_str());
```
Comment (JeongJunSik) [inline external/avs-device-sdk/AVSCommon/Utils/src/Logger/ConsoleLogger.cpp]: please check tab size and remove empty space

File: external/avs-device-sdk/AVSCommon/Utils/src/LibcurlUtils/LibcurlUtils.cpp (hunk 1 above)
Comment (zhouxinhe) [reply external/avs-device-sdk/AVSCommon/Utils/src/LibcurlUtils/LibcurlUtils.cpp]: fixed

File: apps/examples/avs_test/avs_test_main.cpp (hunk 2 above)
Comment (zhouxinhe) [reply apps/examples/avs_test/avs_test_main.cpp]: fixed, thanks

**Comment by seinfra** [comment]
Target : [c0fb1a168c73b537796022975bb181d67e41f98b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c0fb1a168c73b537796022975bb181d67e41f98b] - Code Rule Check (C++) OK.

**Comment by zhouxinhe** [comment]
avs_device_sdk is added as a submodule (#2575 ), so close this PR.

**Pull request closed without merging**